baybe/utils/sampling_algorithms.py

```diff
@@ -13,51 +13,65 @@ def farthest_point_sampling(
 
-    Creates a subset of a collection of points by successively adding points with the
-    largest Euclidean distance to intermediate point selections encountered during
-    the algorithmic process.
+    Creates a subset of a given collection of points by successively adding points with
+    the largest Euclidean distance to intermediate point selections encountered during
+    the algorithmic process. The mechanism used for the initial point selection is
+    configurable.
 
     Args:
-        points: The points that are available for selection, represented as a 2D array
-            whose first dimension corresponds to the point index.
+        points: The points that are available for selection, represented as a 2-D array
+            of shape (n, k), where n is the number of points and k is the dimensionality
+            of the points.
         n_samples: The total number of points to be selected.
-        initialization: Determines how the first points are selected. When
-            ``"farthest"`` is chosen, the first two selected points are those with the
-            largest distance. If only a single point is requested, it is selected
-            randomly from these two. When ``"random"`` is chosen, the first point is
-            selected uniformly at random.
+        initialization: Determines how the first points are selected:
+            * ``"farthest"``: The first two selected points are those with the
+              largest distance. If only a single point is requested, a deterministic
+              choice is made based on the point coordinates.
+            * ``"random"``: The first point is selected uniformly at random.
 
     Returns:
         A list containing the positional indices of the selected points.
 
     Raises:
-        ValueError: If an unknown initialization recommender is used.
+        ValueError: If the provided array is not two-dimensional.
+        ValueError: If the array contains no points.
+        ValueError: If the input space has no dimensions.
+        ValueError: If an unknown method for initialization is specified.
     """
-    # Compute the pairwise distances between all points
+    if np.ndim(points) != 2:
+        raise ValueError("The provided array must be two-dimensional.")
+    if (n_points := len(points)) == 0:
+        raise ValueError("The provided array must contain at least one row.")
+    if points.shape[-1] == 0:
+        raise ValueError("The provided input space must be at least one-dimensional.")
+
+    # Sort the points to produce the same result regardless of the input order
+    sort_idx = np.lexsort(tuple(points.T))
+    points = points[sort_idx]
+
+    # Pre-compute the pairwise distances between all points
     dist_matrix = pairwise_distances(points)
 
-    # Initialize the point selection subset
+    # Avoid wrong behavior pathological situation where all points are duplicates
+    np.fill_diagonal(dist_matrix, -np.inf)
```

Comment:
Couldn't we check this in the very beginning? If all points are just duplicates of each other, I'd have assumed that ew give a warning since this might be an error and just return the point?

Reply:
Having duplicate points is not a problem per se, i.e. it is a valid input (though not very useful to apply FPS in that sense). But still, the algorithm should be able to select a sequence without complaining. So this line is really just to avoid that the subsequent code lines fail – no warning or similar necessary, I think. Feel free to either close or let me know your thoughts

Reply:
Fully agree that this is not a problem, just wanted to make the suggestion that if we have such a degenerate case, we can save some computation (e.g. `pairwise_distance`) by instead checking for this case at the top of the method. In my opinion, this feels a bit more natural than a manual adjustment of the matrix

Reply:
But then what? If you check and don't adjust the matrix, then you need to run an alternative logic instead. You mean that in this case we should just return `list(range(n_points))`? Could do 🤔 Or what is your suggestion?

Reply:
Yeah something like that. But I have the feeling that you prefer to have only a single bit of logic wich is also fine for me :)

Reply:
No, it's fine. Normally I'd say having this as an extra step that requires additional computation (i.e. check for uniqueness) would not warrant the change, but I thought once again and think that perhaps throwing a warning is not a bad a idea here. This is really something we should actually never encounter in bug-free operation, so having a warning that avoids letting this case pass silently should be fine. Have made the necessary changes and added the test for it. Feel free to close if happy